### PR TITLE
MDEV-16144 Default TIMESTAMP clause for SELECT from versioned

### PR DESCRIPTION
**Pull request by midenok**
@abarkov Please review part about type handler.

This code is submitted under the BSD-new license.

### CODE REVIEW

File: sql/table.h [hunk 1]
```diff
@@ -1832,21 +1833,9 @@ class Vers_history_point : public vers_history_point_t
...
-    return false;
-  }
-  bool resolve_unit_timestamp(THD *thd)
-  {
-    if (unit == VERS_UNDEFINED)
-      unit= VERS_TIMESTAMP;
-    return false;
-  }
+  bool check_unit(THD *thd);
   void bad_expression_data_type_error(const char *type) const;
```
Comment (vuvova) [inline sql/table.h]: Do you still need this declaration?

File: sql/sql_type.h [hunk 2]
```diff
@@ -979,6 +978,62 @@ class Information_schema_character_attributes
 };
 
 
+enum vers_kind_t
+{
+  VERS_UNDEFINED= 0,
```
Comment (vuvova) [inline sql/sql_type.h]: What do you still need VERS_UNDEFINED for?

Comment (midenok) [reply sql/sql_type.h]: It means table is unversioned.

File: sql/table.h (hunk 1 above)
Comment (midenok) [reply sql/table.h]: Fixed and rebased on latest 10.3.

**Comment by midenok** [comment]
Moved to [bb-10.5-midenok](https://github.com/MariaDB/server/commits/bb-10.5-midenok).

**Pull request closed without merging**